Seymour/Matroid/Graphicness.lean

```diff
@@ -175,3 +175,115 @@ theorem Matroid.IsGraphic.isRegular {M : Matroid α} (hM : M.IsGraphic) :
     M.IsRegular := by
   obtain ⟨X, Y, A, hA, hMA⟩ := hM
   exact ⟨X, Y, A, hA.isTotallyUnimodular, hMA⟩
+
+
+open scoped BigOperators Matrix
+
+variable {X Y E ι₁ ι₂ 𝔽 : Type*}
+variable [Field 𝔽]
+
+/-- Dot product on `E → 𝔽` using a finite sum. -/
+def dot [Fintype E] (v w : E → 𝔽) : 𝔽 :=
+  ∑ e, v e * w e
+
+def rowSpace {ι E 𝕂 : Type*} [Field 𝕂] (A : Matrix ι E 𝕂) : Submodule 𝕂 (E → 𝕂) :=
+  Submodule.span 𝕂 (Set.range (fun i => A i))
+
+/-- Orthogonal complement (with respect to `dot`) of a submodule of `E → 𝔽`. -/
+def orth [Fintype E] (U : Submodule 𝔽 (E → 𝔽)) : Submodule 𝔽 (E → 𝔽) :=
```

Comment:
It might already exist with some API in Mathlib or in Peter Nelson's repository.